src/Microsoft.ML.Data/Transforms/ValueMappingTransformer.cs

```diff
@@ -0,0 +1,995 @@
+                var valueGetter = cursor.GetGetter<TValue>(1);
+                while(cursor.MoveNext())
+                {
+                    TKey key = default;
+                    TValue value = default;
+                    keyGetter(ref key);
+                    valueGetter(ref value);
+                    if (_mapping.ContainsKey(key))
+                    {
+                        throw env.Except($"Duplicate keys in data '{key}'");
+                    }
+                    _mapping.Add(key, value);
+                }
+            }
+
+            public override Delegate GetGetter(Row input, int index)
+            {
+                var src = default(TKey);
+                ValueGetter<TKey> getSrc = input.GetGetter<TKey>(index);
+                ValueGetter<TValue> retVal =
+                (ref TValue dst) =>
+                {
+                    getSrc(ref src);
+                    if (_mapping.ContainsKey(src))
+                    {
+                        if (ValueType.IsVector)
+                            dst = Utils.MarshalInvoke(GetVector<int>, ValueType.ItemType.RawType, _mapping[src]);
+                        else
+                            dst = Utils.MarshalInvoke(GetValue<int>, ValueType.RawType, _mapping[src]);
+                    }
+                    else
+                       dst = _missingValue;
+                };
+                return retVal;
+            }
+
+            public override IDataView GetDataView(IHostEnvironment env)
+                => DataViewHelper.CreateDataView(env,
+                                                 _mapping.Keys,
+                                                 _mapping.Values,
+                                                 ValueMappingTransformer.KeyColumnName,
+                                                 ValueMappingTransformer.ValueColumnName,
+                                                 ValueType.IsKey);
+
+            private static TValue GetVector<T>(TValue value)
+            {
+                if (value is VBuffer<T> valueRef)
+                {
+                    VBuffer<T> dest = default;
+                    valueRef.CopyTo(ref dest);
+                    if (dest is TValue destRef)
+                        return destRef;
+                }
+
+                return default;
+            }
+
+            private static TValue GetValue<T>(TValue value)
+                => value;
+        }
+
+        /// <summary>
+        /// Retrieves the byte array given a dataview and columns
+        /// </summary>
+
+        private static byte[] GetBytesFromDataView(IHost host, IDataView lookup, string keyColumn, string valueColumn)
+        {
+            Contracts.AssertValue(host);
+            host.AssertValue(lookup);
+            host.AssertNonEmpty(keyColumn);
+            host.AssertNonEmpty(valueColumn);
+
+            var schema = lookup.Schema;
+
+            if (!schema.TryGetColumnIndex(keyColumn, out int colKey))
```

Comment:
>TryGetColumnIndex [](start = 24, length = 17)

if you rebase to latest this is gone in favor of GetColumnOrNull

Reply:
Thanks for letting me know

---
In reply to: [243495500](https://github.com/dotnet/machinelearning/pull/1710#discussion_r243495500) [](ancestors = 243495500)